### PR TITLE
Add a new generator config to change `PrinterColumn` sort function

### PR DESCRIPTION
**Pull request by A-Hilaly**
Issue https://github.com/aws-controllers-k8s/community/issues/823

This patch introduce a new generator configuration under resource
configuration, mainly to tune the `kubebuilder:printcolumn` marker
comments generation.

More precisely we add `PrintConfig` struct that contains one field
called `OrderBy` used to intruct the code generator how to sort the
`kubebuilder:printcolumn` marker comments.

e.g configuration:
```yaml
resources:
  Deployment:
    print:
      order_by: JSONPath
```

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

**Comment by A-Hilaly** [comment]
NOTE to reviewers: This branch need to be rebased after #82 merge.
/hold

**Review by jaypipes — APPROVED** [review]
Nice. ++

**Pull request merged**
